btree/btree.c

```diff
@@ -5128,24 +5160,75 @@ static void btree_ut_thread_kv_oper(struct btree_ut_thread_info *ti)
 							       &kv_op),
 						 &kv_op.bo_sm_group,
 						 &kv_op.bo_op_exec);
+			if (get_data.flags == M0_BSC_KEY_BTREE_BOUNDARY)
+				break;
+
 			keys_found_count++;
 
 			/** Copy over the gotten key for the next search. */
-			for (i = 1; i < ARRAY_SIZE(key); i++)
-				key[i] = get_key[0];
+			for (i = 0; i < ARRAY_SIZE(key); i++)
+				key[i] = get_key[i];
 		}
 
 		M0_ASSERT(keys_found_count == keys_put_count);
 
+		/**
+		 *  Test slant only if possible. If the increment counter is
+		 *  more than 1 we can provide the intermediate value to be got
+		 *  in slant mode.
+		 */
+
+		if (ti->ti_key_incr > 1) {
+			uint64_t  slant_key;
+			uint64_t  got_key;
+
+			M0_ASSERT(key_first >= 1);
+
+			slant_key = key_first - 1;
+			get_data.check_value = false;
+
+			do {
+				key[0] = (slant_key <<
+					  (sizeof(ti->ti_thread_id) * 8)) +
+					 ti->ti_thread_id;
+				key[0] = m0_byteorder_cpu_to_be64(key[0]);
+				for (i = 1; i < ARRAY_SIZE(key); i++)
+					key[i] = key[0];
+
+				M0_BTREE_OP_SYNC_WITH_RC(&kv_op.bo_op,
+							 m0_btree_get(tree,
+								      &rec.r_key,
+								      &ut_get_cb,
```

Comment:
80 char limit exceed for line 5200, 5201. Please check

Reply:
This is difficult for me to do because everything is shifted to the right. I was hoping no one would see it. Anyways let me try some changes.